File: src/main/java/hzt/aoc/day23/LinkedNode.java
package hzt.aoc.day23;

public class LinkedNode<T> {

    private T value;
    private LinkedNode<T> next;

    public LinkedNode() {
        this(null);
    }

    public LinkedNode(T value) {
        this.setValue(value);
    }

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        this.value = value;
    }

    public LinkedNode<T> getNext() {
        return next;
    }

    public void setNext(LinkedNode<T> next) {
        this.next = next;
    }

    @Override
    public String toString() {
        return "CustomLinkedNode{" + getValue() + '}';
    }
}

File: src/main/kotlin/hzt/aoc/day23/LinkedNode.kt
package hzt.aoc.day23

class LinkedNode<T> @JvmOverloads constructor(value: T? = null) {
    var value: T? = null
    var next: LinkedNode<T>? = null
    override fun toString(): String {
        return "CustomLinkedNode{$value}"
    }

    init {
        this.value = value
    }
}
